src/types/privatekubectl.ts

```diff
@@ -0,0 +1,121 @@
+import { Kubectl } from "./kubectl";
+import { ExecOptions, ExecOutput, getExecOutput } from "@actions/exec";
+import * as core from "@actions/core";
+import * as os from "os";
+import * as fs from "fs";
+
+
+
+export class PrivateKubectl extends Kubectl{
+
+
+  public isPrivate(): boolean {
+      return true;
+  }
+
+  protected async execute(args: string[], silent: boolean = false) {
+    args.unshift("/k8stools/kubectl");
+    var kubectlCmd = args.join(" ");
+    var addFileFlag = false;
+    var eo = <ExecOptions>({ silent });
+
+    if(this.containsFilenames(kubectlCmd)){
+      core.debug("kubectl command contains filenames: " + kubectlCmd);
+      kubectlCmd = kubectlCmd.replace(/[\/][t][m][p][\/]/g ,"");
+      core.debug("Removing leading slashes for filenames when invoking for private clusters: " + kubectlCmd);
+      addFileFlag = true;
+    }
+
+      const privateClusterArgs = ["aks", "command", "invoke", 
+      "--resource-group", this.resourceGroup, 
+      "--name", this.name,
+      "--command", kubectlCmd 
+    ];
+    
+    if(addFileFlag){
+      var filenames = this.extractFilesnames(kubectlCmd).split(" ");
+      const tempDirectory = process.env["runner.tempDirectory"] || os.tmpdir() + "/manifests";
+      eo.cwd = tempDirectory;
+      core.debug("ExecOptions current working directory: " + eo.cwd);
+      privateClusterArgs.push(...["--file", "."]);
+
+      var filenamesArr = filenames[0].split(",");
```

Comment:
use const here too. Use it everywhere the variable doesn't change and let everywhere the variable will change